service/http.proxy/src/main/java/org/kaazing/gateway/service/http/proxy/HttpProxyServiceHandler.java

```diff
@@ -232,8 +253,9 @@ private static void processRequestHeaders(HttpAcceptSession acceptSession, HttpC
             }
         }
 
-        // Add Via: 1.1 kaazing header
-        connectSession.addWriteHeader(HEADER_VIA, VIA_HEADER_VALUE);
+        // Add Via: 1.1 kaazing + authority header
+        String authority = acceptSession.getLocalAddress().getExternalURI().getAuthority();
+        connectSession.addWriteHeader(HEADER_VIA, VIA_HEADER_VALUE + SEPARATOR + authority);
```

Comment:
I would suggest create via header value using some guid in the constructor of HttpProxyServiceHandler.

```
    private final String viaHeader;

    HttpProxyServiceHandler() {
        viaHeader = String.format("1.1 kaazing-%s", UUID.randomUUID());
    }
```